assets/js/googlesitekit/datastore/user/adblocker.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * `core/user` data store: adblocker.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { detectAnyAdblocker } from 'just-detect-adblock';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from './constants';
+
+// Actions
+const CHECK_ADBLOCKER = 'CHECK_ADBLOCKER';
+const RECEIVE_IS_ADBLOCKER_ACTIVE = 'RECEIVE_IS_ADBLOCKER_ACTIVE';
+
+export const initialState = {
+	isAdBlockerActive: undefined,
+};
+
+export const actions = {
+	*checkAdBlocker() {
+		return yield {
+			payload: {},
+			type: CHECK_ADBLOCKER,
+		};
+	},
+	receiveIsAdBlockerActive( isAdBlockerActive ) {
+		invariant(
+			'boolean' === typeof isAdBlockerActive,
+			'isAdBlockerActive must be boolean.'
+		);
+		return {
+			payload: { isAdBlockerActive },
+			type: RECEIVE_IS_ADBLOCKER_ACTIVE,
+		};
+	},
+};
+
+export const controls = {
+	[ CHECK_ADBLOCKER ]: async () => {
+		if ( await detectAnyAdblocker() ) {
+			return true;
+		}
+		// The above is good about detecting most adblockers.
+		// For the rest, we'll make a placeholder request to the favicon with some
+		// additional stuff in the query string to (hopefully) trigger a filter.
+		// If this throws, then the fetch request failed completely and we'll assume it was blocked.
+		try {
+			const params = [
+				// The name of the parameter here doesn't really matter
+				// since adblockers look at the URL as a whole.
+				// Note: this value must not be URL-encoded.
+				'google-site-kit=/adsense/pagead2.googlesyndication.com/pagead/js/adsbygoogle.js',
+				// Add a timestamp for cache-busting.
+				`timestamp=${ Date.now() }`,
+			];
+			await fetch( `/favicon.ico?${ params.join( '&' ) }`, {
+				credentials: 'omit',
+				// Don't follow any redirects; we only care about this request being blocked or not.
+				redirect: 'manual',
+			} );
+		} catch {
+			return true;
+		}
+		return false;
```

Comment:
I know this is copied from before, but let's add some breathing room while we're here 😉 
```suggestion

		// The above is good about detecting most adblockers.
		// For the rest, we'll make a placeholder request to the favicon with some
		// additional stuff in the query string to (hopefully) trigger a filter.
		// If this throws, then the fetch request failed completely and we'll assume it was blocked.
		try {
			const params = [
				// The name of the parameter here doesn't really matter
				// since adblockers look at the URL as a whole.
				// Note: this value must not be URL-encoded.
				'google-site-kit=/adsense/pagead2.googlesyndication.com/pagead/js/adsbygoogle.js',
				// Add a timestamp for cache-busting.
				`timestamp=${ Date.now() }`,
			];
			await fetch( `/favicon.ico?${ params.join( '&' ) }`, {
				credentials: 'omit',
				// Don't follow any redirects; we only care about this request being blocked or not.
				redirect: 'manual',
			} );
		} catch {
			return true;
		}

		return false;
```